sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -247,15 +247,22 @@ public bool IsRunning
 
         /// <summary>
         ///   Interacts with the storage system with responsibility for creation of checkpoints and for ownership claim.
+        ///   Lazily evaluated.
         /// </summary>
         ///
-        private PartitionManager StorageManager { get; }
+        private Lazy<PartitionManager> StorageManager { get; }
```

Comment:
Sure you don't want to go with your original thought on the constructor?    I don't have a strong opinion for getting through release, but that's probably the better design the more I think through it.

Reply:
I will update it then. Not a big deal.

Reply:
I was referring to the `PartitionManager` via constructor approach.   I agree that there isn't an option that I think is great, but that seems to be the lesser evil.   Not something we have to do now, but we probably should think it over once we cut the package so that we can tune for post-preview.6.